winch/codegen/src/codegen/context.rs

```diff
@@ -236,7 +236,18 @@ impl<'a, 'builtins> CodeGenContext<'a, 'builtins> {
     }
 
     /// Prepares arguments for emitting a binary operation.
-    pub fn binop<F, M>(&mut self, masm: &mut M, size: OperandSize, mut emit: F)
+    pub fn binop<F, M>(&mut self, masm: &mut M, size: OperandSize, emit: F)
+    where
+        F: FnMut(&mut M, Reg, Reg, OperandSize),
+        M: MacroAssembler,
+    {
+        let dst = self.binop_impl(masm, size, emit);
+        self.stack.push(dst.into());
+    }
+
+    /// Prepares arguments for emitting a binary operation and returns
+    /// destination register.
+    fn binop_impl<F, M>(&mut self, masm: &mut M, size: OperandSize, mut emit: F) -> TypedReg
```

Comment:
I wasn't really sure what to name this. I wanted a method that would do everything `binop` did except push the value on to the stack so I could insert an operation to perform the type change.